### ISSUE TITLE
CA1508: False negative using DoesNotReturnIf

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn-analyzers#4743]
```vb
Imports System.Diagnostics.CodeAnalysis

Public Module Program
    Public Sub Main(args As String())
        Test(args.Length = 0)
        If args.Length = 0 Then ' Expected diagnostic here.
        End If
    End Sub

    Private Sub Test(<DoesNotReturnIf(False)> x As Boolean)
        If Not x Then
            Throw New Exception()
        End If

        Console.WriteLine("Hello, world")
    End Sub
End Module
```

If `Test` is conditionally compiled, there shouldn't be a diagnostic (see #4742). Both issues are closely related, I'd expect both of them to be fixed together.

**Comment by mavasani** [comment]
Tagging @genlu - we may want to write a unit test with interprocedural analysis is enabled. Even then, I am not sure this is a critical case for the analyzer.

**Comment by mavasani** [comment]
Closing this out as it is unlikely we are going to invest further in interprocedural analysis enhancements due to performance concerns.
